### PR TITLE
fhir formatter test

### PR DESCRIPTION
**Pull request by Loui4**

### CODE REVIEW

File: src/common/utils/fhir/formatter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  lastUpdated: null,
+};
+
+describe('fhir formatter', () => {
+  describe('getProductsSystems', () => {
+    const getElement = jest.spyOn(formatterModule, 'getProductSystems');
+    it('should retrieve systems from product dto', () => {
+      const result = formatterModule.getProductSystems(productDto);
+      expect(result).toEqual(expect.arrayContaining(['a', 'b']));
+      expect(getElement).toHaveBeenCalled();
```
Comment (danielmwakanema) [inline src/common/utils/fhir/formatter.spec.ts:24]: By virtue of the prior test passing it implies _getProductSystems_ was called.

File: src/common/utils/fhir/formatter.spec.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  mappings: [
+    { systemName: 'a', systemProductCode: 'code-a', productName: 'prod-a' },
+    { systemName: 'b', systemProductCode: 'code-b', productName: 'prod-b' },
+  ],
+  dateCreated: null,
+  lastUpdated: null,
+};
+
+describe('fhir formatter', () => {
+  describe('getProductsSystems', () => {
```
Comment (danielmwakanema) [inline src/common/utils/fhir/formatter.spec.ts:19]: Perhaps you should also add a test to see if it responds correctly to scenarios in which no products have been returned from the master.

File: src/common/utils/fhir/formatter.spec.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        [productDto],
+        ['a', 'b'],
+      );
+
+      expect(getElement).toHaveBeenCalled();
+      expect(result.group[0].target).toBe('a');
+      expect(result.group[1].target).toBe('b');
+    });
+  });
+});
```
Comment (danielmwakanema) [inline src/common/utils/fhir/formatter.spec.ts:96]: The tests seem fine at the moment. Just be sure to add tests that assert that the code responds well to edge cases like an empty list of products fetched from the master etc.

**Review by danielmwakanema — CHANGES_REQUESTED** [review]

File: src/common/utils/fhir/formatter.spec.ts (hunk 1 above)
Comment (Loui4) [reply src/common/utils/fhir/formatter.spec.ts:24]: yeah thanks for pointing that out, i made a mistake

File: src/common/utils/fhir/formatter.spec.ts (hunk 2 above)
Comment (Loui4) [reply src/common/utils/fhir/formatter.spec.ts:19]: alright.

File: src/common/utils/fhir/formatter.spec.ts (hunk 3 above)
Comment (Loui4) [reply src/common/utils/fhir/formatter.spec.ts:96]: great. I will work on that

**Review by danielmwakanema — APPROVED** [review]
Looks good to me at the moment.
